examples/triangle_intersection/triangle_intersection.cpp

```diff
@@ -322,9 +322,9 @@ int main()
 
   // Create BVH tree
   ArborX::BasicBoundingVolumeHierarchy<
-      MemorySpace, ArborX::Details::PairIndexVolume<Box>> const
+      MemorySpace, ArborX::Details::PairValueIndex<Box>> const
       tree(execution_space,
-           ArborX::Details::LegacyValues<decltype(triangles), Box>{triangles});
+           ArborX::Details::AttachIndices<decltype(triangles)>{triangles});
```

Comment:
OK but noting this still bad to advertise using private implementation details in an example.
I am not expecting you to address this in this PR.

Reply:
`AttachIndices` should really be public. I was not sure whether to do this in this PR. If we agree on the name, I'll do it.